server/src/main/java/com/linecorp/centraldogma/server/internal/api/converter/ChangesRequestConverter.java

```diff
@@ -80,28 +80,34 @@ private static Change<?> readChange(JsonNode node) {
         checkArgument(node.get("path") != null && node.get("type") != null,
                       "a change should have a path and a type");
         final ChangeType changeType = ChangeType.parse(node.get("type").textValue());
+        final JsonNode content = node.get("content");
         if (changeType != ChangeType.REMOVE) {
-            checkArgument(node.get("content") != null, "a change should have a content.");
+            checkArgument(content != null, "a change should have a content.");
         }
 
         final String path = node.get("path").textValue();
         if (changeType == ChangeType.UPSERT_TEXT) {
-            return Change.ofTextUpsert(path, node.get("content").textValue());
+            return Change.ofTextUpsert(path, content.textValue());
         }
         if (changeType == ChangeType.UPSERT_JSON) {
-            return Change.ofJsonUpsert(path, node.get("content"));
+            if (content.isTextual()) {
+                // A content can be a serialized JSON so the text value needs to be parsed as JSON.
+                return Change.ofJsonUpsert(path, content.textValue());
```

Comment:
Question: Can we apply this approach that checks if it's textual in the `Change.ofJsonUpsert` method?

Reply:
`"string"` is a valid JSON type. So I don't think we can generally apply it to `Change.ofJsonUpsert()`.
This is a special case where a web application sends a stringfied JSON content after editing to the server.